File: ruby/testcode/benchmark_test_00975.rb
require_relative 'shared'

def delete_own_post(req)
  post_id = req.param('id')
  current_user = req.cookie('user_id')
  post_owner = "user_#{post_id.to_i % 10}"
  return BenchmarkResponse.error('forbidden', 403) unless post_owner == "user_#{current_user}"
  BenchmarkResponse.ok("post #{post_id} deleted")
end

File: rust/testcode/benchmark_test_00975.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let path = req.param("path");
    let full_url = format!("https://example.com/{}", path);
    let location = format!("Location: {}", full_url);
    super::shared::BenchmarkResponse::ok(&location)
}
